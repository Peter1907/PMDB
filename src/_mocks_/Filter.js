/* eslint-disable*/
import { useState } from 'react';
import s from './Filter.module.css';

export default function Filter() {
  const [category, setCategory] = useState('none');

  const handleSelection = (e) => {
    setCategory(e.target.value);
    e.target.nextSibling.value = '';
    console.log('working');
  };

  const handleChange = (e) => {
    if (category !== 'none') {
      console.log('working');
    }
  };

  return (
    <div className={s.filter}>
      <p>Search By:</p>
      <select className={s.options} onChange={(e) => handleSelection(e)}>
        <option value="none">No-Filters</option>
        <option value="title">Title</option>
        <option value="year">Year</option>
        <option value="imDbRating">Rating</option>
      </select>
      <input className={s.search} type="text" onChange={(e) => handleChange(e)} />
    </div>
  );
}
