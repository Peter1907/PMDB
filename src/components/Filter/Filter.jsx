import { useState } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import s from './Filter.module.css';

export default function Filter({ filter, remove }) {
  const dispatch = useDispatch();
  const [category, setCategory] = useState('none');

  const handleSelection = (e) => {
    setCategory(e.target.value);
    e.target.nextSibling.value = '';
    dispatch(remove());
  };
  
  const handleChange = (e) => {
    if (category !== 'none') {
      dispatch(filter(category, e.target.value));
    }
  };

  return (
    <div className={s.filter}>
      <p>Search By:</p>
      <select className={s.options} onChange={(e) => handleSelection(e)}>
        <option value="none">No-Filters</option>
        <option value="title">Title</option>
        <option value="year">Year</option>
        <option value="imDbRating">Rating</option>
      </select>
      <input className={s.search} type="text" onChange={(e) => handleChange(e)}/>
    </div>
  );
}

Filter.propTypes = {
  filter: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
}
