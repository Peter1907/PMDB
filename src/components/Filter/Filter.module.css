.filter {
  padding: 10px;
  display: flex;
  align-items: center;
  gap: 0.5rem;
  width: 60rem;
  margin: auto;
  justify-content: flex-end;
}

.options,
.search {
  width: 10rem;
  border-radius: 0;
  border: 1px solid #222831;
  font-family: 'Cairo', sans-serif;
  padding: 5px 7px;
  background-color: #eee;
  box-shadow: 1px 1px 10px 1px #22283162;
}
