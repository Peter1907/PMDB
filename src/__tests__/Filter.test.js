/* eslint-disable*/
import renderer from 'react-test-renderer';
import Filter from '../__mocks__/Filter';

it('renders correctly', () => {
  const tree = renderer
    .create(<Filter />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
