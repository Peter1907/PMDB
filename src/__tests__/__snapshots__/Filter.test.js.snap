// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="filter"
>
  <p>
    Search By:
  </p>
  <select
    className="options"
    onChange={[Function]}
  >
    <option
      value="none"
    >
      No-Filters
    </option>
    <option
      value="title"
    >
      Title
    </option>
    <option
      value="year"
    >
      Year
    </option>
    <option
      value="imDbRating"
    >
      Rating
    </option>
  </select>
  <input
    className="search"
    onChange={[Function]}
    type="text"
  />
</div>
`;
